File: src/main/java/engineer/thomas_werner/euler/Problem25.java
package engineer.thomas_werner.euler;

import engineer.thomas_werner.euler.math.FibonacciSequence;

import java.math.BigInteger;
import java.util.function.Predicate;

public class Problem25 {

    public static void main(String[] args) {
        final Problem25 problem25 = new Problem25();
        System.out.println(problem25.findFibonacciIndex(bi -> bi.toString().length() >= 1_000));
    }

    public int findFibonacciIndex(final Predicate<BigInteger> predicate) {
        int index = 1;
        final FibonacciSequence sequence = new FibonacciSequence();
        while(!predicate.test(sequence.nextValue()))
            index++;
        return index;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem25.kt
package de.huddeldaddel.euler

import de.huddeldaddel.euler.math.FibonacciSequence

import java.math.BigInteger
import java.util.function.Predicate

/**
 * Solution for https://projecteuler.net/problem=25
 */
fun main() {
    val problem25 = Problem25()
    println(problem25.findFibonacciIndex(Predicate { bi -> bi.toString().length >= 1000 }))
}

class Problem25 {

    fun findFibonacciIndex(predicate: Predicate<BigInteger>): Int {
        var index = 1
        val sequence = FibonacciSequence()
        while (!predicate.test(sequence.nextValue()))
            index++
        return index
    }

}
